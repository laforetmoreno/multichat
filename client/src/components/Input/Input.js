import React from 'react';
import { string, func } from 'prop-types';

import styles from './Input.module.scss';

const Input = ({ name, onChange, value, placeholder, className }) => (
  <input
    className={`${styles.wrapper} ${className}`}
    name={name}
    onChange={onChange}
    value={value}
    placeholder={placeholder}
  />
);

Input.defaultProps = {
  placeholder: '',
  className: '',
};

Input.propTypes = {
  name: string.isRequired,
  onChange: func.isRequired,
  value: string.isRequired,
  placeholder: string,
  className: string,
};

export default Input;
