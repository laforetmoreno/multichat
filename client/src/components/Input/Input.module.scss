@import '../../styles/variables';

.wrapper,
.error {
  padding: 10px;
  border-radius: 4px;
}

.wrapper {
  font-size: 16px;
  outline: none;
  width: calc(100% - 20px);
  background: $input-background;
  border: none;

  &:hover,
  &:focus {
    border-color: $primary-color;
    transition: all 0.3s;
  }
}

.error {
  color: $error-color-font;
  background: $error-color;
  margin-bottom: 10px;
  font-size: 12px;
}
