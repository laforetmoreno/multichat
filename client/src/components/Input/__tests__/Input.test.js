import React from 'react';
import Input from '../Input';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('<Input />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Input />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a input with placeholder", () => {
    const tree = renderer
    .create(<Input placeholder='Exemple' />)
    expect(tree).toMatchSnapshot();
  });

  it('get by id and change', async () => {
    const { getByTestId } = render(<Input />)
      const fieldNode = await waitForElement(() => getByTestId('input'))
      fireEvent.change(fieldNode, { target: { value: 'test change' } })
  });
});