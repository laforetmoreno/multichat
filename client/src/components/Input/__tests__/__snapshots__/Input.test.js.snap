// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<input
  className="wrapper "
  data-testid="input"
  placeholder=""
/>
`;

exports[`<Input /> should render a input with placeholder 1`] = `
<input
  className="wrapper "
  data-testid="input"
  placeholder="Exemple"
/>
`;
